components/BarcodeScanner.vue

```diff
@@ -98,8 +102,8 @@ export default {
         const selectedCamera = scanner.getRunningTrackSettings()?.deviceId;
         updateSelectedCamera(selectedCamera);
       }
-      // Default to turning light on
-      if(!cameraLight.value) {
+      // Turn on torch by default, if requested
+      if(cameraLight.value != props.lightTorch) {
```

Comment:
Hmm, comment doesn't seem to match what this does. It seems like it will set the camera light value to be consistent with whatever the `lightTorch` property is set to.

```suggestion
      // sync camera light value with requested 
      if(cameraLight.value !== props.torchOn) {
```